haskellPackages: update stackage and hackage

expipiplus1 opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #172337. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

expipiplus1 committed:
all-cabal-hashes: 2022-05-10T13:45:20Z -> 2022-05-14T01:13:33Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

erictapen committed:
haskellPackages.squeal-postgresql: fix build

neilmayhew committed:
haskellPackages.debian: Fix test failures for ghc 9

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #173196 from neilmayhew/debian-haskell

haskellPackages.debian: Fix test failures for ghc 9

cdepillabout committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

TeofilC committed:
haskellPackages.threadscope: unmark as broken, remove jailbreak

cdepillabout committed:
Merge pull request #173250 from TeofilC/unmark-broken-threadscope

haskellPackages.threadscope: Unmark threadscope as broken

neilmayhew committed:
haskellPackages.polynomial: Fix build failures for ghc 9

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages: fix package pins

maralorn committed:
haskellPackages.hspec-core: bump version pin

maralorn committed:
haskellPackages.hspec-discover: bump version pin

github-actions[bot] committed:
Merge master into haskell-updates

neilmayhew committed:
haskellPackages: Remove polynomial from the broken list

neilmayhew committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/packages-marked-as-broken-should-come-with-an-explanation/19187/5

cdepillabout committed:
Merge pull request #173328 from neilmayhew/haskell-polynomial

haskellPackages.polynomial: Fix build failures for ghc 9

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
release-haskell.nix: include nixos tests for xmonad

I've recently discovered they were broken by a recent XMonad upgrade,
something we should keep tabs on in the future.

sternenseemann committed:
nixos/tests/xmonad: adjust to XMonad 0.17

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

maralorn committed:
haskell.packages.ghc922.hls-ormolu-plugin: jailbreak

maralorn committed:
haskellPackages.sensei: fix build

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.fast-tags: apply patch for nondeterministic tests

Resolves #172843.

sternenseemann committed:
Merge pull request #173576 from sternenseemann/xmonad-tests

[haskell-updates] release-haskell.nix: include nixos tests for xmonad

maralorn committed:
Merge pull request #173740 from sternenseemann/fast-tags-nondeterminism

haskellPackages.fast-tags: apply patch for nondeterministic tests

maralorn committed:
maintainers/scripts/haskell/hydra-report.hs: fix outdated hydra-unstable reference

maralorn committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

maralorn committed:
maintainers/scripts/haskell/hydra-report.hs: remove redundant pragmas

maralorn committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

maralorn committed:
Merge branch 'master' into haskell-updates
